src/index.js

```diff
@@ -480,6 +490,89 @@ export default class AuthnWidget {
     document.querySelector('#submit').disabled = disabled;
   }
 
+  async registerIdVerificationRequiredEventHandler() {
+    let data = this.store.getStore();
+
+    if (data.errorDetails !== undefined)
+    {
+      document.getElementById("requiredDescription").style.display = "none";
+      document.getElementById("errorDescription").style.display = "block";
+
+      document.getElementById("errorBlock").style.display = "flex";
+      document.getElementById("errorMessage").innerHTML = this.makeIdVerificationErrorMessage(data.errorDetails);
+    }
+    else
+    {
+      document.getElementById("requiredDescription").style.display = "block";
+      document.getElementById("errorDescription").style.display = "none";
+
+      document.getElementById("errorBlock").style.display = "none";
+    }
+
+    const options = {
+      method: 'GET'
+    }
+    let qrUrlRespone = await fetch(data.qrUrl, options);
+    let qrCode = await qrUrlRespone.text();
+    document.getElementById('qrCode').src = qrCode;
+    document.getElementById('qrCodeBlock').style.display = 'block';
+
+    var verificationCodeToken = data.verificationCode.match(/.{1,4}/g);
+    document.getElementById('verificationCode').innerHTML = verificationCodeToken.join(' ');
+  }
+
+  postIdVerificationRequired() {
+    document.getElementById('copy')
+            .addEventListener('click', this.copyCode);
+    this.pollAction(5000);
+  }
+
+  handleIdVerificationInProgress() {
+    this.pollAction(5000);
+  }
+
+  handleIdVerificationFailed() {
+    let data = this.store.getStore();
+
+    if (data.errorDetails !== undefined)
+    {
+      document.getElementById("errorMessage").innerHTML = this.makeIdVerificationErrorMessage(data.errorDetails);
+    }
+  }
+
+  copyCode(event) {
+    event.preventDefault();
+    let range = document.createRange();
+    range.selectNode(document.getElementById('verificationCode'));
+    window.getSelection().removeAllRanges();
+    window.getSelection().addRange(range);
+    document.execCommand('copy');
+    window.getSelection().removeAllRanges();
+  }
+
+  pollAction(timeout) {
+    setTimeout(() => {
+      this.store.dispatch('POST_FLOW', 'poll', '{}');
```

Comment:
Can we get errors back from this POST_FLOW? If not I think we could go directly to fetchutil.postFlow() method and do the checking for a change here. Once the code changes, we would dispatch a GET_FLOW to re-render. This might help get rid of the specific state code in render.

Reply:
Changed based on what you suggested, please take a look, thanks

Reply:
I really like the changes. Looks good to me. Thanks!